### PR TITLE
Implement `-dumpversion` compiler argument

### PR DESCRIPTION
**Pull request by twwwt**
#6258 

Implement `-dumpversion` compiler argument.

As the changes that come with this pull request are rather simple, they haven't been tested, to be honest. Cross-check please.

### CODE REVIEW

**Comment by kripken** [comment]
The patch looks good, thanks, but the test errors like relevant: the failing tests are ones that use configure or cmake (freetype, zlib, etc.). So I suspect that they are somehow confused by emcc accepting `-dumpversion` or by its output. The best way to debug it might be to run `emconfigure ./configure` in say `tests/zlib` (or a copy in a fresh location) and see what error it runs into.

**Comment by twwwt** [comment]
I will check. Need to find some time first.

File: emcc.py [hunk 1]
```diff
@@ -379,6 +379,10 @@ def run():
   elif '-dumpmachine' in sys.argv:
     print(shared.get_llvm_target())
     exit(0)
+  
+  elif '-dumpversion' in sys.argv: # gcc's doc states "Print the compiler version [...] and don’t do anything else."
```
Comment (echeese) [inline emcc.py]: It looks like python is tripping up over the apostrophe in this comment

Comment (twwwt) [reply emcc.py]: Sorry, my editor's autocorrection ... modern times, uff.

Will someone try without that apostrophe, or should I update the pull request?

Comment (kripken) [reply emcc.py]: I think it's easiest to update the PR with another commit and let tests run here.

**Comment by kripken** [comment]
Great, thanks!

Please just add yourself to AUTHORS before we merge.

**Pull request merged**